events/api/build.gradle.kts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.smallrye.openapi.api.OpenApiConfig.OperationIdStrategy
+import io.smallrye.openapi.gradleplugin.SmallryeOpenApiExtension
+import io.smallrye.openapi.gradleplugin.SmallryeOpenApiTask
+
+plugins {
+  `java-library`
+  jacoco
+  `maven-publish`
+  signing
+  `nessie-conventions`
+  alias(libs.plugins.smallrye.openapi)
+}
+
+extra["maven.name"] = "Nessie - Events - API"
+
+dependencies {
+  implementation(project(":nessie-model"))
+
+  // Immutables
+  implementation(libs.immutables.builder)
+  implementation(libs.immutables.value.annotations)
+  annotationProcessor(libs.immutables.value.processor)
+
+  // Open API
+  implementation(libs.microprofile.openapi)
+
+  // Jackson
+  implementation(platform(libs.jackson.bom))
+  implementation(libs.jackson.annotations)
+  implementation(libs.jackson.databind)
+
+  // ErrorProne
+  implementation(libs.errorprone.annotations)
+
+  // Testing
+  testImplementation(platform(libs.junit.bom))
+  testImplementation(libs.bundles.junit.testing)
+  testImplementation(libs.junit.jupiter.engine)
+  testImplementation(libs.jackson.datatype.jdk8)
+  testImplementation(libs.jackson.datatype.jsr310)
+}
+
+extensions.configure<SmallryeOpenApiExtension> {
+  scanDependenciesDisable.set(false)
+  infoVersion.set(project.version.toString())
+  schemaFilename.set("META-INF/openapi/openapi")
+  operationIdStrategy.set(OperationIdStrategy.METHOD)
+  scanPackages.set(listOf("org.projectnessie.events.spi"))
+  scanClasses.set(
+    listOf(
+      "org.projectnessie.model.Content",
+      "org.projectnessie.model.ContentKey",
+      "org.projectnessie.model.DeltaLakeTable",
+      "org.projectnessie.model.IcebergTable",
+      "org.projectnessie.model.IcebergView",
+      "org.projectnessie.model.Namespace",
+    )
+  )
+}
+
+val openapiSource: Configuration by
+  configurations.creating {
+    isCanBeConsumed = true
+    isCanBeResolved = false
+    description = "Source OpenAPI spec files, containing the examples"
+  }
+
+val generateOpenApiSpec =
+  tasks.named<SmallryeOpenApiTask>("generateOpenApiSpec") {
+    inputs.files("src/main").withPathSensitivity(PathSensitivity.RELATIVE)
+  }
+
+artifacts { add(openapiSource.name, file("src/main/resources/META-INF")) }
```

Comment:
I guess this would need to be published to swaggerhub as well - and added as a release artifact.
Aka: changes to the `release-publish.yml` workflow

Reply:
Ah good point.

Taking a step back, I think I'm going to remove the OpenAPI bits. All I wanted was for users to have a Json schema to use, e.g. in Kafka. But we don't need a full-blown OpenAPI spec here, since there is no operations in this package (no REST endpoints.)

Searching a bit, it seems that many Kafka users are solving this problem using this project: https://github.com/mbknor/mbknor-jackson-jsonSchema . It can generate Json schemas from POJOs and honors all Jackson annotations, including `@JsonSubTypes`.

So I suggest that we remove OpenAPI for now, and when I get to the reference implementation, we can reassess the need for a Json schema here.

Reply:
That project looks unmaintained, the project license is okay though.

But yes, removing the openapi bits for events is fine.
Guess it makes sense to move the "schema concerns" to when we get to the reference implementation.